### PR TITLE
feat(wasm): Oracle support

### PR DESCRIPTION
**Pull request by joss-aztec**
Wasm bindings for solving the intermediate witness with a callback for resolving oracle opcodes.

### CODE REVIEW

File: common/src/serializer.rs [hunk 1]
```diff
@@ -341,16 +341,19 @@ pub fn serialize_circuit(circuit: &Circuit) -> ConstraintSystem {
             Opcode::Directive(_) => {
                 // Directives are only needed by the pwg
             }
-            Opcode::Block(_, _) => {
+            Opcode::Block(_) => {
                 // TODO: implement serialization of blocks to match BB's interface
             }
+            Opcode::ROM(_) | Opcode::RAM(_) | Opcode::Oracle(_) => {
+                todo!("Unimplemented")
```
Comment (joss-aztec) [inline common/src/serializer.rs:348]: @vezenovm, I'm guessing Block/ROM/RAM will be addressed separately as part of your ultraplonk work.

I guess it should be possible to serialise oracle opcodes now though. Reckon you'll have time to give me a had with that?

Comment (joss-aztec) [reply common/src/serializer.rs:348]: Thinking about it, I reckon oracles don't actually add any constraints. I'll stub it alongside the `Directive` case.

Comment (vezenovm) [reply common/src/serializer.rs:348]: > I'm guessing Block/ROM/RAM will be addressed separately as part of your ultraplonk work.

Yes you can leave these unimplemented for now

> Thinking about it, I reckon oracles don't actually add any constraints. I'll stub it alongside the `Directive` case.

You are correct here as well, they should follow what we do for directives. The constraints will be laid down elsewhere in ACIR gen

**Comment by TomAFrench** [comment]
Closing this PR as the `Oracle` opcode no longer exists.

**Pull request closed without merging**